### PR TITLE
Use explicit buffer and texture bindings on shaders

### PR DESCRIPTION
**Pull request by gdkchan**
This change uses explicit binding points on the shader code, rather than calling the OpenGL `glGet*Location` functions with the texture/image/buffer name and assigning a binding point on the backend. The main advantage here is making the backend cleaner and more flexible.

Advantages:
- Code is simpler/smaller.
- Binding should be a little bit faster, as it no longer needs to get the binding point from index/stage, the binding point is passed directly. A few other things were optimized aswell (for example, it passes all buffers to be bound at once instead of one at a time).

Other changes:
- Buffers are properly unbound instead of just keeping the last binding.

I recommend testing on a few games just to be sure nothing regressed.

### CODE REVIEW

**Review by Thog — APPROVED** [review]
lgtm, it surely help a lot to simplify some logics :+1: 
Still need to do more testing on games

**Comment by EmulationFanatic** [comment]
Seems to work fine in my testing

File: Ryujinx.Graphics.Gpu/Engine/Methods.cs [hunk 1]
```diff
@@ -1005,6 +1009,10 @@ private void UpdateShaderState(GpuState state)
 
                 if (info == null)
                 {
+                    TextureManager.SetGraphicsTextures(stage, Array.Empty<TextureBindingInfo>());
+                    TextureManager.SetGraphicsImages(stage, Array.Empty<TextureBindingInfo>());
+                    BufferManager.SetGraphicsStorageBufferBindings(stage, null);
+                    BufferManager.SetGraphicsUniformBufferBindings(stage, null);
                     continue;
```
Comment (Leischii) [inline Ryujinx.Graphics.Gpu/Engine/Methods.cs:1016]: New line missing

**Review by riperiperi — APPROVED** [review]
Very nice! No objections here.

**Pull request merged**